tests/test_build_html.py

```diff
@@ -147,7 +147,7 @@ def check_extra_entries(outdir):
 @pytest.mark.sphinx('html', testroot='warnings')
 def test_html_warnings(app, warning):
     app.build()
-    html_warnings = strip_escseq(warning.getvalue().replace(os.sep, '/'))
+    html_warnings = strip_escseq(re.sub(re.escape(os.sep) + '{1,2}', '/', warning.getvalue()))
```

Comment:
I found back the line in code that generates the repr-string and causes the `\\` pattern to exist.  It's sphinx/directives/code.py, line 216:
```
raise UnicodeError(_('Encoding %r used for reading included file %r seems to '
                     'be wrong, try giving an :encoding: option') %
                   (self.encoding, filename))
```

Based on this vs. the other error lines in this test, it appears that some of the warning messages are inconsistent about using `%s` vs. `%r` for printing file paths.  In theory this could be cleaned up in the code, but it seems like for now modifying the test to accept it is a better solution and has fewer unknown consequences.